wal/wal.go

```
@@ -265,10 +266,12 @@ func (w *WAL) ReadAll() (metadata []byte, state raftpb.HardState, ents []raftpb.
 			return nil, state, nil, fmt.Errorf("unexpected block type %d", rec.Type)
 		}
 	}
-	if err != io.EOF {
+	// we must read all of the entries if WAL is opened in write mode.
+	if err != io.EOF && w.f != nil {
```

why do we ignore the error when in read mode?